### ISSUE TITLE
Chart fails to install 

### ISSUE THREAD

**Issue by sekka1** [SumoLogic/sumologic-kubernetes-collection#985]
I am trying to install the sumologic helm chart version `1.3.0` into a kubernetes cluster `v1.16.14` with helmv3.

Here are the values I am setting the chart with:
```
    falco:
      enabled: true
    prometheus-operator:
      kubeProxy:
        enabled: false
      kubelet:
        serviceMonitor:
          https: false
      nodeExporter:
        enabled: true
      prometheus-node-exporter:
        service:
          port: 49100
          targetPort: 49100
      prometheusOperator:
        enabled: false
    sumologic:
      clusterName: kubernetes-dev-us
      envFromSecret: sumo-access-key-secret
```

Nothing gets created b/c the initial setup job fails:
```
2wire609-v:ht-devops garlandkan$ kubectl -n monitoring describe jobs
Name:           sumologic-sumologic-setup
Namespace:      monitoring
Selector:       controller-uid=3b8f2aaf-2bfb-4f22-8f50-6e3aeacbe2a4
Labels:         app=sumologic-sumologic
                chart=sumologic-1.3.0
                heritage=Helm
                release=sumologic
Annotations:    helm.sh/hook: pre-install,pre-upgrade
                helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
                helm.sh/hook-weight: 3
Parallelism:    1
Completions:    1
Pods Statuses:  0 Running / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           controller-uid=3b8f2aaf-2bfb-4f22-8f50-6e3aeacbe2a4
                    job-name=sumologic-sumologic-setup
  Service Account:  sumologic-sumologic-setup
  Containers:
   setup:
    Image:      sumologic/kubernetes-fluentd:1.3.0
    Port:       <none>
    Host Port:  <none>
    Command:
      /etc/terraform/setup.sh
    Limits:
      cpu:     2
      memory:  256Mi
    Requests:
      cpu:     200m
      memory:  64Mi
    Environment Variables from:
      sumo-access-key-secret  Secret  Optional: false
    Environment:              <none>
    Mounts:
      /etc/terraform from setup (rw)
  Volumes:
   setup:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      sumologic-sumologic-setup
    Optional:  false
Events:
  Type     Reason        Age                 From            Message
  ----     ------        ----                ----            -------
  Warning  FailedCreate  51s (x4 over 2m1s)  job-controller  Error creating: pods "sumologic-sumologic-setup-" is forbidden: error looking up service account monitoring/sumologic-sumologic-setup: serviceaccount "sumologic-sumologic-setup" not found
```

The service account it is looking for does not exist.  

Why doesnt the chart create the service account it needs?

**Comment by frankreno** [comment]
@sekka1 - the ServiceAccount should be created (https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/master/deploy/helm/sumologic/templates/setup/setup-serviceaccount.yaml).  Can you please share some more details:

1. Please share the command you used to install
2. What version of Kubernetes? 
3. What platform (eg managed service like EKS/GKE/AKS or self managed)

**Comment by sekka1** [comment]
I think i found the problem.  It looks like the `serviceaccount` is created in the `default` namespace and not in the namespace where the release is.

Here is a PR with the changes: https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/986

**Comment by frankreno** [comment]
@sekka1 thanks for the PR! However, I believe there is something else going on here because none of our templates have the `namespace` specified as Helm provides this via flags. Are you using helm2 or helm3? When you install with helm2 you can use the `--namespace` flag to install the release in a specific namespace and if it does not exist it will create it.  In helm3 you can use the `--namespace` flag and `--create-namespace` if the namespace does not exist.

I am running the following to install and it is creating the ServiceAccount in the correct namespace. Note the `monitoring` namespace did not exist.

<img width="864" alt="Screen Shot 2020-10-06 at 9 03 15 PM" src="https://user-images.githubusercontent.com/1776470/95282816-6fecec00-0817-11eb-89d2-01585eddb95f.png">

**Comment by sekka1** [comment]
I am using helm v3.
I am using helm v3 and installing this with Weaveworks Flux.  Perhaps it is doing something different as the install command.

It is not totally accurate that none of the templates specifies the namespace:
The setup job has a namespace in it: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/master/deploy/helm/sumologic/templates/setup/setup-job.yaml#L6

**Comment by sekka1** [comment]
Here is my values file for the `helmrelease`:

```
---
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: sumologic
  namespace: monitoring
spec:
  interval: 1h0m0s
  releaseName: sumologic
  targetNamespace: monitoring
  chart:
    spec:
      chart: sumologic
      version: 1.2.2
      interval: 1m
      sourceRef:
        kind: HelmRepository
        name: sumologic
        namespace: gotk-system
  values:
    # Falco settings: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/master/deploy/helm/sumologic/values.yaml#L1376
    falco:
      enabled: true
    # This is using the kube-prometheus-stack chart
    # https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml
    prometheus-operator:
      prometheusOperator:
        enabled: false
      nodeExporter:
        enabled: true
      prometheus-node-exporter:
        # If you have another prometheus node exporter running, it is probably on port 9100.  This sets
        # The port to another port so they dont collide
        service:
          port: 49100
          targetPort: 49100      
      kubelet:
        serviceMonitor:
          https: false
      kubeProxy:
        enabled: false
```

For some reason the stuff in the `setup` folder is going to the `default` namespace instead.

**Comment by pmalek-sumo** [comment]
I'm trying to reproduce it on my cluster and I could see the `serviceaccount`:

```
kubectl get serviceaccounts -n monitoring
NAME                        SECRETS   AGE
default                     1         4h54m
sumologic-sumologic-setup   1         3m58s
```

and the job itself:

```
kubectl describe jobs -n monitoring
Name:           sumologic-sumologic-setup
Namespace:      monitoring
Selector:       controller-uid=326a542c-ec10-45e6-8dcc-9b127d82f16c
Labels:         app=sumologic-sumologic
                chart=sumologic-1.2.3
                heritage=Helm
                release=sumologic
Annotations:    helm.sh/hook: pre-install,pre-upgrade
                helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
                helm.sh/hook-weight: 3
Parallelism:    1
Completions:    1
Start Time:     Mon, 12 Oct 2020 16:02:52 +0200
Pods Statuses:  1 Running / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           controller-uid=326a542c-ec10-45e6-8dcc-9b127d82f16c
                    job-name=sumologic-sumologic-setup
  Service Account:  sumologic-sumologic-setup
  Containers:
   setup:
    Image:      sumologic/kubernetes-fluentd:1.2.3
    Port:       <none>
    Host Port:  <none>
    Command:
      /etc/terraform/setup.sh
    Environment:
      SUMOLOGIC_ACCESSID:   x
      SUMOLOGIC_ACCESSKEY:  x
      SUMOLOGIC_BASE_URL:
      HTTP_PROXY:
      HTTPS_PROXY:
    Mounts:
      /etc/terraform from setup (rw)
  Volumes:
   setup:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      sumologic-sumologic-setup
    Optional:  false
Events:
  Type    Reason            Age    From            Message
  ----    ------            ----   ----            -------
  Normal  SuccessfulCreate  3m37s  job-controller  Created pod: sumologic-sumologic-setup-ggn2s
```

 created in `monitoring` namespace, when using the `gotk` setup that can be observed below (as well as [those instructions](https://toolkit.fluxcd.io/get-started/))

```yaml
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: HelmRepository
metadata:
  name: sumologic
  namespace: gotk-system
spec:
  interval: 1m
  url: https://sumologic.github.io/sumologic-kubernetes-collection
---
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: sumologic
  namespace: monitoring
spec:
  interval: 1m0s
  releaseName: sumologic
  targetNamespace: monitoring
  chart:
    spec:
      chart: sumologic
      version: 1.2.3
      interval: 1m
      sourceRef:
        kind: HelmRepository
        name: sumologic
        namespace: gotk-system
  values:
    sumologic:
      accessId: "x"
      accessKey: "x"
    falco:
      enabled: false
    prometheus-operator:
      prometheusOperator:
        enabled: false
      nodeExporter:
        enabled: true
      prometheus-node-exporter:
        # If you have another prometheus node exporter running, it is probably on port 9100.
        # This sets the port to another port so they dont collide
        service:
          port: 49100
          targetPort: 49100
      kubelet:
        serviceMonitor:
          https: false
      kubeProxy:
        enabled: false
```

Can you maybe share a bit more info on how you have your setup configured?

**Comment by sekka1** [comment]
@pmalek-sumo I had the same config you posted above and was shocked that it worked for you =).  I was also surprised that you went and did this with `gotk`.  Thanks!  Since your config is the same as mine, i would have imagined you went with the latest `gotk` release.  I was using `v0.1.0`.  There seems to be a few updates since then.  I changed my version to `v0.1.5` and it worked.  I did have to manually delete a bunch of stuff like pod security policies, clusterroles, clusterrolebindings, and even some sumo services in the `kube-system` namespace but after deleting all of those, it came up!

Thanks for helping me out =)
